includes/class-monk-links.php

```diff
@@ -0,0 +1,448 @@
+<?php
+/**
+ * The engine to change links according to language
+ *
+ * @since      0.2.0
+ *
+ * @package    Monk
+ * @subpackage Monk/Links
+ */
+
+/**
+ * Class to hold all functions related to links
+ * the permalinks are filtered to return content due to languages
+ *
+ * @package    Monk
+ * @subpackage Monk/Links
+ */
+class Monk_Links {
+
+	/**
+	 * The ID of this plugin.
+	 *
+	 * @since    0.1.0
+	 * @access   private
+	 * @var      string    $monk    The ID of this plugin.
+	 */
+	private $monk;
+
+	/**
+	 * The version of this plugin.
+	 *
+	 * @since    0.1.0
+	 * @access   private
+	 * @var      string    $version    The current version of this plugin.
+	 */
+	private $version;
+
+	/**
+	 * Initializes the class and set its properties.
+	 *
+	 * @since    1.0.0
+	 * @param      string $monk       The name of the plugin.
+	 * @param      string $version    The version of this plugin.
+	 */
+	public function __construct( $monk, $version ) {
+		$structure = get_option( 'permalink_structure', false );
```

Comment:
Now the class has a method called 'monk_using_permalinks()'
Retrieves the structure or false, if the user is not using pretty permalinks
Have I found a good approach?